module-2/backend/week_two.md

```diff
@@ -26,20 +39,84 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+```ruby
+team = Team.find(4)
+team.owner
+
+```
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+<img src="http://i.imgur.com/MYu34Aj.png">
+
 6. Define foreign key, primary key, and schema.
+  * A foreign key is a field in one table that refers to the primary key located in another table. A primary key is the object's `id`. A schema shows the current state of the database. When you create a table (`rake db:create migration NAME=<table_name>`) and migrate it (`rake db:migrate`), the schema will update accordingly.
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+  * A foreign key on one table indicates its' relationship with the primary key on another table (it can be a one-to-many, one-to-one, or many-to-many relationship). Using the example above, a team can belong to an owner and an owner can have many teams. This is a one-to-many relationship. When we specifically indicate this relationship in the `Team` and `Owner` class, we're instructing Ruby to create a foreign key column on the `Team` table using the unique primary key located in the `Owner` table.
```

Comment:
Careful here: indicating the relationship on the `Team` and `Owner` class doesn't tell Ruby to create a foreign key column; our migration does that. Indicating the relationship at the class level tells Ruby/ActiveRecord to **look for/use** the foreign key (which exists because of our migrations) to find related rows of data.